### ISSUE TITLE
Face Correction

### ISSUE THREAD

**Issue by cmdr2** [cmdr2/stable-diffusion-ui#49]
GFPGAN Face Correction seems like a decent start

**Comment by MrManny** [comment]
Just to let the fine folks know, this has landed in the `beta` branch.
See the beta announcement on Discord: https://discord.com/channels/1014774730907209781/1017700054007558226/1017830307304575056

**Comment by cmdr2** [comment]
It is now on the `main` branch, so anyone can use it after installing (or restarting the software).

Keeping this open until it's confirmed that it's working for people.

**Comment by oyajiru** [comment]
Hi, while running start.sh, I get the following error:

```
/home/user/Applications/stable-diffusion-ui/stable-diffusion/env/lib/python3.8/site-packages/torchvision/io/image.py:13: UserWarning: Failed to load image Python extension: libtorch_cuda_cu.so: cannot open shared object file: No such file or directory
  warn(f"Failed to load image Python extension: {e}")
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/user/Applications/stable-diffusion-ui/stable-diffusion/src/gfpgan/gfpgan/__init__.py", line 2, in <module>
    from .archs import *
  File "/home/user/Applications/stable-diffusion-ui/stable-diffusion/src/gfpgan/gfpgan/archs/__init__.py", line 2, in <module>
    from basicsr.utils import scandir
  File "/home/user/Applications/stable-diffusion-ui/stable-diffusion/env/lib/python3.8/site-packages/basicsr/__init__.py", line 3, in <module>
    from .archs import *
  File "/home/user/Applications/stable-diffusion-ui/stable-diffusion/env/lib/python3.8/site-packages/basicsr/archs/__init__.py", line 5, in <module>
    from basicsr.utils import get_root_logger, scandir
  File "/home/user/Applications/stable-diffusion-ui/stable-diffusion/env/lib/python3.8/site-packages/basicsr/utils/__init__.py", line 5, in <module>
    from .img_util import crop_border, imfrombytes, img2tensor, imwrite, tensor2img
  File "/home/user/Applications/stable-diffusion-ui/stable-diffusion/env/lib/python3.8/site-packages/basicsr/utils/img_util.py", line 6, in <module>
    from torchvision.utils import make_grid
  File "/home/user/Applications/stable-diffusion-ui/stable-diffusion/env/lib/python3.8/site-packages/torchvision/__init__.py", line 7, in <module>
    from torchvision import models
  File "/home/user/Applications/stable-diffusion-ui/stable-diffusion/env/lib/python3.8/site-packages/torchvision/models/__init__.py", line 18, in <module>
    from . import quantization
  File "/home/user/Applications/stable-diffusion-ui/stable-diffusion/env/lib/python3.8/site-packages/torchvision/models/quantization/__init__.py", line 1, in <module>
    from .mobilenet import *
  File "/home/user/Applications/stable-diffusion-ui/stable-diffusion/env/lib/python3.8/site-packages/torchvision/models/quantization/mobilenet.py", line 1, in <module>
    from .mobilenetv2 import QuantizableMobileNetV2, mobilenet_v2, __all__ as mv2_all
  File "/home/user/Applications/stable-diffusion-ui/stable-diffusion/env/lib/python3.8/site-packages/torchvision/models/quantization/mobilenetv2.py", line 5, in <module>
    from torch.ao.quantization import QuantStub, DeQuantStub
ModuleNotFoundError: No module named 'torch.ao'


Dependency test failed! Error installing the packages necessary for GFPGAN (Face Correction).
```

**Comment by pixartist** [comment]
Does not seem to work. I ran it and it certainly does not perform face correction

**Comment by jwax33** [comment]
Ran the same prompt/seed without it and with it enabled.  The two faces are virtually identical but there are some very tiny differences.  It's working, but whether that's working as intended I can't say.

**Comment by cmdr2** [comment]
@pixartist @jwax33 It really depends on the prompt. It'll be effective for some prompts, while not for others. I've added the same face-correction library that every other Stable Diffusion software uses, so that's a limitation of the model used.

For e.g. try this prompt:

Prompt: A digital Illustration of a Thomas Metzinger, 4k, detailed, trending in artstation, fantasy vivid colors
Seed: 1817
Guidance Scale: 10.1

Closing this issue since it is working for most users based on replies and artwork shared on Discord. Thanks
